Neos.Neos/Classes/Controller/Module/Management/WorkspacesController.php

```diff
@@ -792,44 +782,60 @@ protected function computeSiteChanges(Workspace $selectedWorkspace, ContentRepos
                 $documentPathSegments = [];
                 foreach ($ancestors as $ancestor) {
                     $pathSegment = $ancestor->nodeName ?: NodeName::fromString($ancestor->nodeAggregateId->value);
-                    $nodePathSegments[] = $pathSegment;
+                    if (!$this->getNodeType($ancestor)->isOfType(NodeTypeNameFactory::NAME_SITES)) {
+                        $nodePathSegments[] = $pathSegment;
+                    }
                     if ($this->getNodeType($ancestor)->isOfType(NodeTypeNameFactory::NAME_DOCUMENT)) {
                         $documentPathSegments[] = $pathSegment;
                         if (is_null($documentNode)) {
                             $documentNode = $ancestor;
                         }
                         // the site node is the last ancestor of type Document
-                        $siteNode = $documentNode;
+                        // TODO: Check for Neos.Neos:Site instead of regular document after it is introduced
+                        $siteNode = $ancestor;
                     }
                 }
 
                 // Neither $documentNode, $siteNode or its cannot really be null, this is just for type checks;
                 // We should probably throw an exception though
                 if ($documentNode !== null && $siteNode !== null && $siteNode->nodeName) {
                     $siteNodeName = $siteNode->nodeName->value;
-                    $documentPath = implode('/', array_slice(array_map(
+                    $documentPath = implode('/', array_reverse(array_map(
                         fn (NodeName $nodeName): string => $nodeName->value,
                         $documentPathSegments
-                    ), 2));
+                    )));
                     $relativePath = str_replace(
                         sprintf('//%s/%s', $siteNodeName, $documentPath),
                         '',
-                        implode('/', array_map(
+                        implode('/', array_reverse(array_map(
                             fn (NodeName $nodeName): string => $nodeName->value,
                             $nodePathSegments
-                        ))
+                        )))
                     );
                     if (!isset($siteChanges[$siteNodeName]['siteNode'])) {
                         $siteChanges[$siteNodeName]['siteNode']
                             = $this->siteRepository->findOneByNodeName(SiteNodeName::fromString($siteNodeName));
                     }
+
                     $siteChanges[$siteNodeName]['documents'][$documentPath]['documentNode'] = $documentNode;
+                    if ($documentNode->equals($node)) {
+                        $siteChanges[$siteNodeName]['documents'][$documentPath]['isNew'] = $change->created;
+                        $siteChanges[$siteNodeName]['documents'][$documentPath]['isMoved'] = $change->moved;
+                    }
```

Comment:
I don't understand this change either. Are those `isNew` and `isMoved` magic strings new?

Reply:
Not exactly. They by the frontend part to visiually show the difference and also for selecting the correct combination of nodes. Thats something we should probably check again and maybe change when we update the workspace module.

Reply:
no, see the removed part further down around (previously 850).